server/channels/app/notification.go

```diff
@@ -124,14 +124,15 @@ func (a *App) SendNotifications(c request.CTX, post *model.Post, team *model.Tea
 	var allActivityPushUserIds []string
 	if channel.Type != model.ChannelTypeDirect {
 		// Iterate through all groups that were mentioned and insert group members into the list of mentions or potential mentions
-		for _, group := range mentions.GroupMentions {
+		for groupID := range mentions.GroupMentions {
+			group := groups[groupID]
```

Comment:
Previously, there was some degree of confidence that `group` would always exist because it was in a sense being returned directly. But now we're trusting that a group was matched and exists in the separate `groups` data structure. Is it worth hardening this to double check that we find a map entry before using it? (Maybe no error, just skip with a warning?)

Reply:
With how closely coupled this code is, I don't think it's necessary here. It's just above this loop where we get `groups` and then pass it into `getExplicitMentionsAndKeywords`, so if there's invalid entries in `mentions.GroupMentions`, I'd be to blame for it